src/server/main_service.cc

```diff
@@ -1167,12 +1167,13 @@ void Service::Exec(CmdArgList args, ConnectionContext* cntx) {
 
 void Service::Publish(CmdArgList args, ConnectionContext* cntx) {
   string_view channel = ArgS(args, 1);
-  string_view message = ArgS(args, 2);
+  // string_view message = ArgS(args, 2);
+  std::shared_ptr<std::string_view> message = std::make_shared<std::string_view>(ArgS(args, 2));
```

Comment:
Can you add a comment for this? I don't understand what the `shared_ptr` is for 😅 It has no custom deleter, you don't check its state anywhere, the type is a string view... 🤔

Reply:
Its not required to have a custom deleter. This is a classic case for using shared ptr. We are populating the message between all the connections that need to publish on the subscribers. We need to make sure that we will not release the message until it is sent. It use to have a shared token to implement it, but in this case we don't need to have it, since the shared_ptr ensure in this case that the message will live long enough until it is published. All this code does is replacing this manual management with what the shared_ptr does. But I will add a comment (I can put back the original ones about the shared token).

Reply:
> This is a classic case for using shared ptr... to make sure that we will not release the message until it is sent

Yes, but the string_view doesn't own the memory. Its a pass-by-value type

Reply:
right - and the shared pointer ensure that we would not let resource go away. I have to say that I wasn't sure myself about the string_view, so initially it was a string (actually const string), but I ran this with multiple connections running at the same time with multiple disconnections and I cannot see any issue. If it would seemed more robust to use string, I will gladly return to it.